tests/utils.py

```diff
@@ -16,6 +16,8 @@
 """ utilities shared between all tests """
 np.random.seed(4)
 
+# function used to generate the data for emulated runs
+DATA_GEN_FN = np.random.random
```

Comment:
Are we planning to use something else here? I think in general it might be a good idea to introduce a fixture that returns a generator, instead of relying on a global `np.random.seed` (considered legacy and might be deprecated at some point). I have something like that for the [autograd plugin tests](https://github.com/flexcompute/tidy3d/blob/develop/tests/test_plugins/autograd/conftest.py), but we might want to tweak and reuse this globally.

Reply:
I agree. and no plans (yours sounds good). I just left it here since it makes it easier for testing. Should I change it? or is it fine to leave like this for now?

Reply:
Was just a general comment, lef‘s leave it for now. Would make sense to add this in one swoop together with the rest of the tests.